cloudify/workflows/local.py

```diff
@@ -545,10 +550,14 @@ def _load_instance(self, node_instance_id):
     def _store_instance(self, node_instance):
         pass
 
-    def get_node_instances(self, node_id=None):
+    def get_node_instances(self, node_id=None, evaluate_functions=True):
         instances = self._node_instances.values()
         if node_id:
             instances = [i for i in instances if i.node_id == node_id]
+        if evaluate_functions:
+            for instance in instances:
+                dsl_functions.evaluate_node_instance_functions(
+                    instance, self)
         return copy.deepcopy(instances)
```

Comment:
Any idea why this returns a deepcopy? It seems like we modify stuff in place and then return a copy for some reason?

Reply:
Just discussed- this result is consumed by plugins, etc, so we need to deepcopy to avoid unexpected mutability issues.